[Game] Added check for Npc and Doodad that there is no same object in…

NL0bP opened the pull request:
… position;
- We currently have Npc and Doodad spawning non-stop. This fix tries to fix that.

NL0bP committed:
[Game] Added check for Npc and Doodad that there is no same object in position;

ZeromusXYZ reviewed (changes requested):
I'll approve this hack, but we might want to mark it as a hack like
`// TODO: This is a hack to prevent infinite spawns until we can find a better fix`

NL0bP committed:
[Game] fixed doodad:
- For doodad: To prevent doodad from infinitely spawning, we will check DoodadFuncQuest and depending on that initialize doodad or not.
- For Npc: check for spawns in the same place.

ZeromusXYZ reviewed (approved):
Also works I guess :)
